### ISSUE TITLE
ImportError: No module named "x" when inherit from Job

### ISSUE THREAD

**Issue by trein** [discoproject/disco#467]
Hi.

I'm experimenting `Disco` and I was trying to create custom reusable jobs by inherit from `Job`. My idea is to create a base class, let's say `MyJob`, which setup a job with some boilerplate configurations as remote server address. What I have so far:

``` python
# mylib/myjob.py
class MyJob(Job):
    # some methods and configuratons

    def execute(self):
        self.run(
            input=self.inputs(),
            save=True,
            sort=True,
        )
        self.outputs()
```

My test job is something taken from Disco's tutorial:

``` python
# wordcount.py
from mylib.myjob import MyJob
from disco.util import kvgroup


class WordCount(MyJob):
    def map(self, line, params):
        for word in line.split():
            yield word, 1

    def reduce(self, iterator, out, params):
        for word, counts in kvgroup(sorted(iterator)):
            yield word, sum(counts)

    def inputs(self):
        return ["http://discoproject.org/media/text/chekhov.txt"]

    def outputs(self):
        for word, count in self.results():
            print(word, count)


if __name__ == '__main__':
    from mylib.myjob import MyJob
    from wordcount import WordCount
    name = 'sample_trein_count'
    WordCount(name).execute()
```

I'm using here the workaround explained [here](https://github.com/discoproject/disco/issues/287) in order to have the runnable code in the same file as the `Job` class.

The problem is that, when I run the job, I get the error:

```
sample_trein_count@573:d46fa:44408:
Status: [map] 0 waiting, 1 running, 0 done, 0 failed
2014/03/26 17:42:18 master     New job initialized!
2014/03/26 17:42:18 master     Created job "sample_trein_count@573:d46fa:44408"
2014/03/26 17:42:18 master     Stage map scheduled with 1 tasks
2014/03/26 17:42:18 master     map:0 assigned to localhost
2014/03/26 17:42:18 master     FATAL: [map:0] Traceback (most recent call last):
  File "/usr/local/var/disco/data/localhost/4f/sample_trein_count@573:d46fa:44408/Library/Python/2.7/site-packages/disco/worker/__init__.py", line 317, in main
    task = cls.get_task()
  File "/usr/local/var/disco/data/localhost/4f/sample_trein_count@573:d46fa:44408/Library/Python/2.7/site-packages/disco/worker/__init__.py", line 402, in get_task
    return Task(**dict((str(k), v) for k, v in cls.send('TASK').items()))
  File "/usr/local/var/disco/data/localhost/4f/sample_trein_count@573:d46fa:44408/Library/Python/2.7/site-packages/disco/task.py", line 70, in __init__
    self.jobobjs = dPickle.loads(self.jobpack.jobdata)
  File "/usr/local/var/disco/data/localhost/4f/sample_trein_count@573:d46fa:44408/Users/trein/wordcount.py", line 1, in <module>
    from mylib.myjob import MyJob
ImportError: No module named mylib.myjob

2014/03/26 17:42:18 master     WARN: Job killed
Status: [map] 1 waiting, 0 running, 0 done, 1 failed
```

**Comment by pooya** [comment]
@trein Is the following line correct?
    from myjob.py import MyJob

**Comment by trein** [comment]
@pooya Sorry for that. Copy and paste mistake. I've fixed the post.

**Comment by trein** [comment]
Hi @pooya,

I've just tested you fix and it solves the problem. However, I had to add my working directory to `PYTHONPATH`. In order to change this behavior, I removed the condition which test if the working directory is in `PYTHONPATH` (line 260 in [modutil.py](https://github.com/discoproject/disco/blob/develop/lib/disco/worker/modutil.py))

``` python
if dirname(job_path) in user_paths():
```

and it works perfectly now and there is no need to change my `PYTHONPATH`. I'm just not quite sure about potential collateral damage caused by this change.

So, I was wondering if it is the expected behavior. If it isn't, does the remotion of such condition bring any side effect?

Thanks,
